src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1680,6 +1683,27 @@ public DatabaseRecord ReadDatabaseRecord()
             }
         }
 
+        public TableSchema GetDocsSchemaForCollection(CollectionName collection) =>
+            _documentsCompression.CompressAllCollections || _compressedCollections.Contains(collection.Name)
+                ? DocumentsStorage.CompressedDocsSchema
+                : DocumentsStorage.DocsSchema;
+
+        private void UpdateCompressionConfigurationFromDatabaseRecord(DatabaseRecord record)
+        {
+            if (_documentsCompression.Equals(record.DocumentsCompression))
```

Comment:
What happens if the configuration changes while there is a running write transaction?
- The same test but starts without compression.
- After deleting the document, we update the compression configuration.
- The update will have the compressed schema while we take the table (from the cache) with the non-compressed schema.

This was also the previous behavior - let's open an issue for that.

Reply:
It's fine to change the compression definition, we handle reading that on a per document basis, after all.
And worst case, we have a document that is compressed or not compressed when it should, but that is just a race condition without an impact.

Reply:
I'm talking about changing it in the **middle** of the transaction.
1. We create 2 documents in a compressed collection.
2. Disable compression.
3. In the same transaction, delete one document and update the second one.
4. Inside the transaction, after we delete the document here:
https://github.com/ravendb/ravendb/pull/17659/files#diff-624ee6c1d1af1fe87f3d169216fa6ea981b9835799020d41b1625cb7bb450ec5R2186
Enable compression.
We already cached the table with non-compressed schema, and now we update the second document when the compression is enabled (which brings us to the same issue that we have in the test).